tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-confusion-matrix.html

```diff
@@ -95,26 +105,32 @@
         },
       },
 
+      observers: [
+        'drawMatrix(counts, allItems)',
+      ],
+
       ready: function() {
         // Allows d3-created elements in this Polymer element to be styled by
         // the css rules above.
         this.scopeSubtree(this.$.holder, true);
       },
 
-      drawMatrix: function(counts) {
+      drawMatrix: function(counts, allItems) {
         this.$.holder.innerHTML = '';
         if (!counts) {
           return;
         }
 
-        // Get all labels from the counts dict.
-        let allLabels = {};
-        const outerLabelList = Object.keys(counts);
-        for (let i = 0; i < outerLabelList.length; i++) {
-          allLabels[outerLabelList[i]] = true;
-          const innerLabelList = Object.keys(counts[outerLabelList[i]]);
-          for (let j = 0; j < innerLabelList.length; j++) {
-            allLabels[innerLabelList[j]] = true;
+        // Get all labels from the provided object, or counts dict.
+        let allLabels = allItems;
```

Comment:
Two points:
1. Below, you are mutating the `allItems` without making a copy. That is dangerous.
2. What you are essentially doing here is creating an array of label list. Just use an array? If there can be dup, please use `const set = new Set()` and do `Array.from(set.values())`

Reply:
changed to using Set and then converting to list as suggested, and removed mutation of provided data.